Month1/Week2/Section1/Exercise3/Main3.cpp

```diff
@@ -0,0 +1,167 @@
+
+/*Exercise 3 - Shapes and Areas
+Create two classes: a Circle class and a Square class. For each class, take in the necessary information to calculate its area and perimeter.  Remember to validate input.
+
+Ask the user if they want to create a circle or a square. For their pick, ask the necessary information to create the shape. Then, tell them the area and the perimeter of their shape.
+
+Your class should have the following methods:
+ExampleShape();
+ExampleShape(/* Params );
+float GetArea();
+float GetPerimeter();
+
+Welcome to create-a-shape, what type of shape do you want?
+1 - Circle
+2 - Square
+Your choice: 1
+
+Enter radius for your circle: 30
+Your shape�s area: 2827.43
+Your shape�s perimeter: 188.5
+
+*/
+
+
+#define _USE_MATH_DEFINES
+#include <iostream>
+#include<string>
+#include <cmath>
+
+class Square
+{
+private:
+	float Side;
+	float AreaSquare;
+	float PerimeterSquare;
+public:
+	Square()
+	{
+	}
+	Square(float Radius, float Diameter)
+	{
+		Side = 0;
+	}
+	void DataSquare()
+	{
+		float side;
+		std::cout << "what is the side of Your Square:" << std::endl;
+		std::cin >> side;
+
+		while (!std::cin.good() || side < 0)
+		{
+			std::cout << "Invalid input, please enter The side of Your square again" << std::endl;
+			std::cin.clear();
+			std::cin.ignore(1000, '\n');
+			std::cin >>side;
+		}
+		Side = side;
```

Comment:
Variables should make logical sense in their operations, that's why they need descriptive names. What would you think is happening here if you only read this line?

Reply:
Done